core/src/main/java/com/linecorp/armeria/client/AbstractRuleBuilder.java

```diff
@@ -29,21 +28,22 @@
 import com.google.common.collect.Iterables;
 import com.google.common.collect.Sets;
 
-import com.linecorp.armeria.client.UnprocessedRequestException;
+import com.linecorp.armeria.client.circuitbreaker.CircuitBreakerRule;
+import com.linecorp.armeria.client.circuitbreaker.CircuitBreakerRuleWithContent;
+import com.linecorp.armeria.client.retry.RetryRule;
+import com.linecorp.armeria.client.retry.RetryRuleWithContent;
 import com.linecorp.armeria.common.HttpStatus;
 import com.linecorp.armeria.common.HttpStatusClass;
 import com.linecorp.armeria.common.RequestHeaders;
 import com.linecorp.armeria.common.ResponseHeaders;
+import com.linecorp.armeria.common.util.UnstableApi;
 
 /**
- * An abstract builder class which creates a {@link RetryRule} or a {@link RetryRuleWithContent}.
+ * An abstract builder class which creates a {@link RetryRule}, a {@link RetryRuleWithContent},
```

Comment:
A skeletal builder implementation for ... and ...? (like `AbstractRuleWithContentBuilder`)